src/tree/AATree.scala

```diff
@@ -0,0 +1,220 @@
+/**
+ * This file is part of Scalacaster project,
+ * https://github.com/vkostyukov/scalacaster and written by Amanj Sherwany,
+ * http://www.amanj.me
+ *
+ * AA tree http://en.wikipedia.org/wiki/AA_tree
+ *
+ * Performance: (From Wikipedia)
+ * The performance of an AA tree is equivalent to the performance of a
+ * red-black tree. While an AA tree makes more rotations than a red-black tree,
+ * the simpler algorithms tend to be faster, and all of this balances out to
+ * result in similar performance. A red-black tree is more consistent in its
+ * performance than an AA tree, but an AA tree tends to be flatter, which
+ * results in slightly faster search times
+ *
+ * Insert - O(log n)
+ * Lookup - O(log n)  
+ * Remove - O(log n)
+ */
+
+object AATree {
```

Comment:
You probably don't need to wrap it into an object. It's easy to read the code w/o it.